lib/rules/no-useless-catch.js

```diff
@@ -43,11 +45,31 @@ module.exports = {
 						context.report({
 							node,
 							messageId: "unnecessaryCatchClause",
+							fix: fixer => fixer.remove(node),
 						});
 					} else {
 						context.report({
 							node: node.parent,
 							messageId: "unnecessaryCatch",
+							fix(fixer) {
+								const tryBlock = node.parent.block;
+								const sourceCode = context.getSourceCode();
+								const firstToken =
+									sourceCode.getFirstToken(tryBlock);
+								const lastToken =
+									sourceCode.getLastToken(tryBlock);
+
+								// Remove the opening and closing braces, keeping everything in between
+								return fixer.replaceText(
+									node.parent,
+									sourceCode
+										.getText()
+										.slice(
+											firstToken.range[1],
+											lastToken.range[0],
+										),
+								);
```

Comment:
If the last statement in the try block is on the same line as the first statement after try-catch, the autofix could be invalid. For example:

```js
/* eslint no-useless-catch: 2 */

try { doSomething() } catch (e) { throw e; } doSomethingMore()
```

After the autofix:

```js
/* eslint no-useless-catch: 2 */

 doSomething()  doSomethingMore() // Parsing error: Unexpected token doSomethingMore

```

Reply:
I think it makes sense to autofix to `{ doSomething() } doSomethingMore()` as per    https://github.com/eslint/eslint/pull/19952#discussion_r2265089073